File: A6vshourie/src/FruitBasket.java
// A6, Varun Shourie, CIS340, Tuesday/Thursday, 3:00PM-4:15PM

import java.util.Scanner;

public class FruitBasket {
	
	// Instantiates a static scanner object for use throughout the program. 
	public static Scanner scanner = new Scanner(System.in);
	
	// Fields which indicate the attributes/contents of each fruit basket object.
	private Apple apple;
	private Banana banana;
	private String basketName;
	
	// Retrieves the basket name for the user. 
	public String getBasketName() {
		return this.basketName;
	}
	
	// Allows the user to alter the name of the fruit basket object.
	public void setBasketName(String basketName) {
		this.basketName = basketName;
	}
	
	// Creates instances/objects of type apple and banana and instantiates their attributes. 
	public void makeFruits() {
		
		// Instantiates an apple with radius of 1.5 inches & a banana object with length of 3.5 inches.
		apple = new Apple(1.5);
		banana = new Banana(3.5);
		
		// Alters the value of peel thickness for both of the fruit objects. 
		apple.setPeelThickness(1);
		banana.setPeelThickness(4);
	}
	
	// Informs the users that there is an apple and banana available, and allows them to consume
	// a portion of each fruit.
	public void eatFruits() {
		// Stores the portion of the fruit the user would like to consume. 
		double amountToEat = 0.0;
	
		// Prints the basket's name and information.
		System.out.printf("\t\t\t***%s***\n", this.basketName.toUpperCase());
		System.out.printf("You have an apple and a banana in your %s basket.\n", this.basketName);
		
		// Prompts the user for the amount of apple they would like to eat.
		System.out.print("What percent of the apple would you like to eat? ");
		amountToEat = Double.parseDouble(scanner.nextLine());
		// Physically consumes the apple by the amount the user has specified.
		apple.eat(amountToEat);
		
		// Prompts the user for the amount of banana they would like to eat.
		System.out.print("What percent of the banana would you like to eat? ");
		amountToEat = Double.parseDouble(scanner.nextLine());
		// Physically consumes the banana by the amount the user has specified. 
		banana.eat(amountToEat);
		
		System.out.println();
		
		// Lets the user know how much of the apple and banana is left.
		System.out.printf("You have %.2f of your apple and %.2f of your banana left in your"
				+ " %s basket.\n", apple.getPercentLeft(), banana.getPercentLeft(), this.basketName);
		
	}

	
}

File: A6vshourieC#/FruitBasket.cs
using System;
using System.Collections.Generic;
using System.Text;
using static System.Console;

namespace A6vshourie
{
    class FruitBasket
    {
        // Store the modeled contents of each fruit basket.
        private Apple apple;
        private Banana banana;
        private string basketName;

        public string GetBasketName()
        {
            return basketName;
        }

        public void SetBasketName(string basketName)
        {
            this.basketName = basketName;
        }

        // Creates modeled apples and bananas and modifies their attributes.
        public void MakeFruits()
        {
            // The arguments provided refer to the length of the fruit.
            apple = new Apple(1.5);
            banana = new Banana(3.5);

            apple.SetPeelThickness(1);
            banana.SetPeelThickness(4);
        }

        // Informs the users of an apple and banana available to consume & allows them to consume a portion of the fruit.
        public void EatFruits()
        {
            // Stores the portion of the fruit the user would like to consume.
            double amountToEat = 0.0;

            Write("\t\t\t***{0}***\n", basketName.ToUpper());
            Write("You have an apple and a banana in your {0} basket.\n", basketName);

            Write("What percent of the apple would you like to eat? ");
            amountToEat = Convert.ToDouble(ReadLine());

            apple.Eat(amountToEat);

            Write("What percent of the banana would you like to eat? ");
            amountToEat = Convert.ToDouble(ReadLine());

            banana.Eat(amountToEat);

            Write("\n");

            Write("You have {0:0.00}% of your apple and {1:0.00}% of your banana left in your" +
                " {2} basket.\n\n", apple.GetPercentLeft(), banana.GetPercentLeft(), basketName);
        }
    }
}
